### ISSUE TITLE
🐛 [Bug Report] Spell Travel Time/Interrupt Bug

### ISSUE THREAD

**Issue by arowjet** [cmangos/issues#2251]
Hi all, got a bug to report with the spell system, if two people spam an instant at each other, ice lance on one char, and counterspell on another char, if the ice lance spell is traveling through the air the caster will receive a full 8 second interrupt on that spell class.
but you are not supposed to be able to "spell school lock" on instant spells or on any spell traveling...even if they DO have a travel time. Once the spell is done casting or if the spell is instant, it shouldn't be interrupted.

thank you for reading, anyone know how to fix possibly? thanks all

btw this was reported by a 10 yr pvp frost mage and I tested it myself on him with a clean core

**Comment by arowjet** [comment]
In SPELLEFFECTS.CPP
void Spell::EffectInterruptCast(SpellEffectIndex /*eff_idx*/)

Change this 
`if (spell->GetCastTime() > 0.0f && (curSpellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_INTERRUPT) && curSpellInfo->PreventionType == SPELL_PREVENTION_TYPE_SILENCE)`

To this

```
if (((spell->getState() == SPELL_STATE_CASTING || spell->getState() == SPELL_STATE_CREATED
				|| spell->getState() == SPELL_STATE_TARGETING || spell->getState() == SPELL_STATE_DELAYED) && spell->GetCastTime() > 0.0f) && (curSpellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_INTERRUPT) && curSpellInfo->PreventionType == SPELL_PREVENTION_TYPE_SILENCE)
```
            {
                unitTarget->LockOutSpells(GetSpellSchoolMask(curSpellInfo), GetSpellDuration(m_spellInfo));
                unitTarget->InterruptSpell(CurrentSpellTypes(i), false);
                interruptedSpellId = curSpellInfo->Id;
            }

**Comment by MantisLord** [comment]
Duplicate of https://github.com/cmangos/issues/issues/2250

Leaving this one open as it has a potential fix in it.

**Comment by Yatzii93** [comment]
Doesn't account for channeled spells.

**Comment by Yatzii93** [comment]
Although this is properly better fixed elsewhere but if you wanted to make your solution work with channeled spells then something like this may be needed. 

` if ((spell->GetCastTime() > 0.0f || (spell->m_spellInfo->HasAttribute(SPELL_ATTR_EX_CHANNELED_1) || spell->m_spellInfo->HasAttribute(SPELL_ATTR_EX_CHANNELED_2))) && (curSpellInfo->InterruptFlags & SPELL_INTERRUPT_FLAG_INTERRUPT) && curSpellInfo->PreventionType == SPELL_PREVENTION_TYPE_SILENCE && spell->CanBeInterrupted())`

**Comment by AnonXS** [comment]
@MantisLord

**Comment by jimmybrancaccio** [comment]
@arowjet Please ensure to fill out _all_ information requested in the bug reporting template. I am closing this out as the other case has much more of the template filled out. This case can be referred to from #2250.
